### PR TITLE
bindinfo: migrate test-infra to testify(staging-3)

### PR DESCRIPTION
**Pull request by w169q169**
### What problem does this PR solve?

Issue Number: ref #27177 

### Release note 

```release-note
None
```

### CODE REVIEW

File: testkit/testkit.go [hunk 1]
```diff
@@ -12,6 +12,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+//go:build !codes
```
Comment (tisonkun) [inline testkit/testkit.go:15]: please revert this.

Comment (w169q169) [reply testkit/testkit.go]: Sorry, the goland in go1.17 convert those flags automatically. I will fix it.

**Comment by w169q169** [comment]
/cc @tisonkun @morgo 
PTAL

**Comment by tisonkun** [comment]
Thanks for your contribution! I'm reviewing now.

**Review by tisonkun — APPROVED** [review]
LGTM

**Review by morgo — APPROVED** [review]

**Pull request merged**